src/blot/scroll.ts

```diff
@@ -16,20 +17,26 @@ const OBSERVER_CONFIG = {
 
 const MAX_OPTIMIZE_ITERATIONS = 100;
 
-class ScrollBlot extends ParentBlot implements Root {
+class ScrollBlot<
+    KnownBlots extends BlotConstructor = BlotConstructor,
```

Comment:
It's an interesting approach. I considered that, but unfortunately, it doesn't work with runtime registrations, which could make it a bit misleading.

I believe we should either refrain from making it generic altogether or disable runtime registration. However, I don't think the latter is feasible.

Reply:
I was thinking that maybe we can keep both and migrate the Quill module API to be static registrations. Runtime registrations is still usable but it won't have TS inference support. Since Quill 2.0 isn't officially released yet, there's opportunity to do breaking changes.

On the other hand, I imaging the use case for actually hot-swapping modules at runtime (dynamic registration) to be scarce.

If it's for lazy-loading, we would still need to wait until blots/attributors are loaded before rendering the editor, otherwise we would potentially get broken content. Then why not have Quill/Parchment statically initialized after modules are loaded?

If it's for dynamically enabling certain features - we can probably have another simple API to disable/enable registry items in runtime.

Reply:
Yeah I agree that allowing only static registrations is a direction worth considering. This would make the code easier to maintain. For Quill, this issue is slightly more troublesome because some blots are registered within another blot, such as `Code` registering `CodeBlockContainer`. We need to think about what kind of API we will provide to address such use cases.

Regardless, I think we need to first determine how we want to modify Quill to implement the static-only registration approach, and if feasible, then change Parchment to the same approach. With that approach, we need to remove `Registry#register()` while adding generics to `Registry`. However this PR modifies the types alone, which does not fully reflect how the Registry works currently, which can be counterintuitive, so I suggest we split this part out of this PR.

Reply:
Done @luin